ingress/controllers/nginx/controller.go

```diff
@@ -797,9 +860,24 @@ func (lbc *loadBalancerController) getPemsFromIngress(data []interface{}) map[st
 		}
 	}
 
+	lbc.secrMetadataLock.Lock()
```

Comment:
@bprashanth I had the feeling that this would be too expensive, but I can modify the code accordingly